t10 api art queries

lanc2112b opened the pull request:
Add sort, order query options.
Add topic filter.  
Refactor selectUsernameByName to correct model.

lanc2112b committed:
Refactor selectUsernameByName into userModel.js.

lanc2112b committed:
T10 - All tests pass
Add sort_by query to /api/articles.
Add order_by query to /api/articles.
Add topic filter to /api/articles.

lanc2112b committed:
remove spare console.log

emilyb93 commented on __tests__/app.test.js:
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
+            expect(article).toMatchObject({
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: expect.any(String),
```
I would say we know that its going to be a topic of 'cats'

lanc2112b replied:
```expect(article.author).toBe('rogersop');
            expect(article.topic).toBe("cats");
```

emilyb93 commented on __tests__/app.test.js:
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
+  describe("GET /api/articles (10) filtered by queries", () => {
+    it("200: Returns an array of objects containing articles filtered by topic > mitch", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch") // has one cat article
```
I think someones been copying their code hmmmm

emilyb93 commented on __tests__/app.test.js:
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
+          articles.forEach((article) => {
+            expect(article).toMatchObject({
+              author: expect.any(String),
+              topic: expect.any(String),
```
We've requested specifically the topic of 'mitch' so i should expect that they all should have that topic

emilyb93 commented on __tests__/app.test.js:
```diff
@@ -127,6 +127,140 @@ describe("GET Endpoints", () => {
+  });
+
+
+
```
Great testing so far, can i just see a test to prove that if i pass order=desc it still works.
Also do you think your code can handle both desc and DESC, will this matter to your api, will your client  have to make this distinction?

emilyb93 reviewed (changes requested):
Just add a test to prove that order=desc is okay to do and have a look at my other comments and then its fine to merge

lanc2112b committed:
Add specific test for mitch topic (l169).
Change selectAllArticles, checks against lower case form for valid order direction.
Add test for order=desc, no other options.

lanc2112b committed:
as previous.
